morpheus/messages/tensor_memory.py

```diff
@@ -31,11 +31,75 @@ class TensorMemory(MessageData, cpp_class=_messages.TensorMemory):
     Parameters
     ----------
     count : int
-        Number of inference inputs.
-    inputs : typing.Dict[str, cupy.ndarray]
-        Inference inputs to model.
+        Length of each tensor contained in `tensors`.
+    tensors : typing.Dict[str, cupy.ndarray]
+        Collection of tensors uniquely identified by a name.
 
     """
     count: int
 
-    tensors: typing.Dict[str, cp.ndarray] = dataclasses.field(default_factory=dict, init=False)
+    tensors: typing.Dict[str, cp.ndarray] = dataclasses.field(default_factory=dict,
+                                                              repr=False,
+                                                              compare=False,
+                                                              hash=False)
+
+    def __init__(self, count: int, tensors: typing.Dict[str, cp.ndarray] = {}):
+        self.count = count
+        self._tensors = tensors
+
+    def get_tensors(self):
+        """
+        Get the tensors contained by this instance. It is important to note that when C++ execution is enabled the
+        returned tensors will be a Python copy of the tensors stored in the C++ object. As such any changes made to the
+        tensors will need to be updated with a call to `set_tensors`.
+
+        Returns
+        -------
+        typing.Dict[str, cp.ndarray]
+        """
+        return self._tensors
+
+    def set_tensors(self, tensors):
+        """
+        Overwrite the tensors stored by this instance. If the length of the tensors has changed, then the `count`
+        property should also be updated.
+
+        Parameters
+        ----------
+        tensors : typing.Dict[str, cupy.ndarray]
+            Collection of tensors uniquely identified by a name.
+        """
+        self._tensors = tensors
+
+    def get_tensor(self, name):
+        """
+        Get the Tensor stored in the TensorMemory container identified by `name`.
+
+        Parameters
+        ----------
+        name : str
+            Tensor key name.
+
+        Returns
+        -------
+        cupy.ndarray
+            Tensor.
+
+        Raises
+        ------
+        KeyError
+            When no matching tensor exists.
+        """
+        return self._tensors[name]
+
+    def set_tensor(self, name, tensor):
+        """
+        Update the tensor identified by `name`. If the length of the tensor has changed, then the `count`
+        property should also be updated.
+
+        Parameters
+        ----------
+        tensors : typing.Dict[str, cupy.ndarray]
+            Collection of tensors uniquely identified by a name.
```

Comment:
Wrong docstring for this argument

Reply:
That isn't an accurate description for `tensors`?
Would this be better:
```
Dictionary of CuPy arrays uniquely identified by name
```